chessmate/engines.py

```diff
@@ -382,6 +382,32 @@ def move(self, board: chess.Board) -> chess.Move:
         return random.choice([*self.legal_moves])
 
 
+class PrioritizePieceMoves(CaptureHighestValue):
+    """ Engine that prioritizes moves for each type of piece in the game,
+    complete with weighting for capture moves """
+
+    def __init__(self, piece):
+        """ See parent docstring """
+        super().__init__()
+        self.name = "Prioritize Piece Moves"
+        self.piece = piece
```

Comment:
Can you add type hinting to this? ```self.piece: str = piece``` would suffice so that we can enforce a string